### PR TITLE
Don't display engine rating for SV engines

### PR DESCRIPTION
**Pull request by SJuliez**
Removes the engine rating (but not the engine type) from the mech summary and the record sheet for support vehicles as they don't use engine ratings.

old:
image

new:
image

### CODE REVIEW

File: megamek/src/megamek/common/Engine.java [hunk 1]
```diff
@@ -450,7 +450,11 @@ public int integralHeatSinkCapacity(boolean compact) {
      */
     public String getShortEngineName() {
         if (engineType < TYPE_KEYS.length) {
-            return String.format("%d%s", engineRating, Messages.getString("Engine." + TYPE_KEYS[engineType]));
+            if (hasFlag(SUPPORT_VEE_ENGINE)) {
+                return String.format("%s", Messages.getString("Engine." + TYPE_KEYS[engineType])).trim();
```
Comment (neoancient) [inline megamek/src/megamek/common/Engine.java:454]: No need to call String.format.
```suggestion
                return Messages.getString("Engine." + TYPE_KEYS[engineType]).trim();
```

**Review by neoancient — CHANGES_REQUESTED** [review]

**Comment by HammerGS** [comment]
Errata issued by CGL that could provide a better option. 

https://bg.battletech.com/forums/index.php/topic,27319.msg1984570.html#msg1984570

**Review by neoancient — APPROVED** [review]

**Comment by neoancient** [comment]
We could change it to show the effective rating for the stackpole rule, but that's easily calculated from the information on the sheet and it only applies to fusion engines. I think it would be fine to leave it off.

**Comment by HammerGS** [comment]
It was originally a CGL request to add the number. They created the errata specifically to show the rating for SV's. So we should go ahead and show the new errata value and engine type for them.

**Pull request merged**